rclcpp/src/rclcpp/client.cpp

```diff
@@ -96,10 +96,9 @@ ClientBase::wait_for_service_nanoseconds(std::chrono::nanoseconds timeout)
       return false;
     }
     node_ptr->wait_for_graph_change(event, time_to_wait);
-    if (event->check_and_clear()) {
-      if (this->service_is_ready()) {
-        return true;
-      }
+    event->check_and_clear();  // reset the event
+    if (this->service_is_ready()) {  // check if the service is ready regardless
```

Comment:
(I know this is captured in the commit message but that can get annoying to track down once the files get moved around)